File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //Get input from the user
    string text = get_string("Text: ");

    //Number of letters
    int letters = count_letters(text);

    //Number of words
    int words = count_words(text);

    //Number of sentences
    int sentences = count_sentences(text);

    //Printing index
    int index = round(0.0588 * ((double)letters * 100 / words) - 0.296 * ((double)sentences * 100 / words) - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int letters = 0;
    //Counting number of letters
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] >= 'A' && text[i] <= 'z')
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    //Counting number of words
    int words = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words + 1;
}

int count_sentences(string text)
{
    //Counting number of sentences
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():

    # Get input from the user
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Calculating index
    index = round(0.0588 * (letters * 100 / words) - 0.296 * (sentences * 100 / words) - 15.8)

    # Evaluating grades
    if index < 1:

        print("Before Grade 1")

    elif index > 16:

        print("Grade 16+")

    else:

        print(f"Grade {index}")


def count_letters(text):
    # Calculating number of letters

    letters = 0
    for i in range(0, len(text)):
        if text[i] >= 'A' and text[i] <= 'z':
            letters += 1
    return letters


def count_words(text):
    # Calculating number of words

    words = 0
    for i in range(0, len(text)):
        if text[i] == ' ':
            words += 1
    return words + 1


def count_sentences(text):
    # Calculating number of sentences

    sentences = 0
    for i in range(0, len(text)):
        if text[i] in ['.', '?', '!']:
            sentences += 1
    return sentences


main()
